lib/formatador/table.rb

```diff
@@ -1,13 +1,13 @@
 class Formatador
-  def display_table(hashes, keys = nil, &block)
+  def display_table(hashes, keys = nil, numeric_rjust: false, &block)
     new_hashes = hashes.inject([]) do |accum,item|
       accum << :split unless accum.empty?
       accum << item
     end
-    display_compact_table(new_hashes, keys, &block)
+    display_compact_table(new_hashes, keys, numeric_rjust: numeric_rjust, &block)
   end
 
-  def display_compact_table(hashes, keys = nil, &block)
+  def display_compact_table(hashes, keys = nil, numeric_rjust: false, &block)
     headers = keys || []
```

Comment:
I noticed that the tests were failing in ruby 3.0 and head. It was a little bit of a struggle initially to figure out what was going on. Since the tests capture stdout, they were also capturing the stack trace. So I commented out the block that does the stdout capture and ran it locally to figure out the specific bug.

 It looks like because keys is an optional argument, and not a named one, if you call something like you did in the tests:
`Formatador.display_table([{'right-justify a numeric' => 12345, 'standard value' => 'value'}], numeric_rjust: true)`
It ends up setting headers to be `numeric_rjust: true`, and then raises an error when it tries to add this to something.

Overall the code seems alright, but we'll need to think about how to tweak this to support what you want. One option that I've done before for similar situations is to replace the old option `keys = nil` with something like `options = nil`. And then you check to see if it is an array or nil to give it the old treatment, or give it a newer treatment if it is a hash. And expect that new usage for keys would also become `{keys: [...]}`. Probably it would also make sense to have a deprecation warning if that was the case. 

Does that make sense? That is at least the first thing to come to mind for me in terms of approach, but I'd be open to other suggestions. It seems like the current way doesn't quite work the way we might like in all situations, so it might need a tweak.

I also tried explicitly passing either `nil` or `[]` for the keys value, along with rjust, but then I get `wrong number of arguments (given 3, expected 1..2)`, which leads me to suspect that the two options wouldn't really play nice with each other in other situations either.

What do you think?

Reply:
Thank you for the detailed explanation.
I’ll think about it.

Reply:
Sounds good.

Another way around this I was thinking about would be to have class level options. So we might have something like:

`Formatador.options[:numeric_rjust] = true`

And then subsequent calls would use that until/unless it was changed again at the class level. That might be a little less messy and tricky to get just right. Or at least something kind of like that could work.

Reply:
It worked, but what do you think?

Reply:
Great idea, I had forgotten about that possibility for fixing it. I think that could really work. Could you maybe add one more test that passes both a value for keys and rjust? I'd like to make sure we cover that case as well and ensure we don't break it in the future. Thanks!

Reply:
Added test that passes both a value for keys and rjust.
Should I rebase it?

Reply:
Looks great, thanks!